examples/connectivity/plot_extract_regions_canica_maps.py

```diff
@@ -0,0 +1,84 @@
+"""
+Regions extraction using Canonical ICA maps and functional connectomes
+======================================================================
+
+This example shows how to use :class:`nilearn.regions.region_extractor.RegionExtractor`
+to extract connected brain regions from whole brain ICA maps and
+use them to estimate a connectome.
+
+We used 20 resting state ADHD functional datasets from :func:`nilearn.datasets.fetch_adhd`
+and :class:`nilearn.decomposition.canica` for whole brain ICA maps.
+
+Please see the related documentation of :class:`nilearn.regions.region_extractor.RegionExtractor`
+for more details.
+"""
+import numpy as np
+from nilearn import datasets
+print(" -- Fetching ADHD resting state functional datasets -- ")
+adhd_dataset = datasets.fetch_adhd(n_subjects=20)
+func_filenames = adhd_dataset.func
+confounds = adhd_dataset.confounds
+
+from nilearn.decomposition.canica import CanICA
+print(" -- Canonical ICA decomposition of functional datasets -- ")
+# Initialize canica parameters
+n_components = 5
+canica = CanICA(n_components=n_components, smoothing_fwhm=6.,
+                memory="nilearn_cache", memory_level=2,
+                threshold=3., random_state=0)
+
+canica.fit(func_filenames)
+components_img = canica.masker_.inverse_transform(canica.components_)
+
+from nilearn.regions import region_extractor
+print(" -- Extracting regions from ICA maps and timeseries signals -- ")
+extractor = region_extractor.RegionExtractor(
+    components_img, threshold="98%", standardize=True, min_size=100)
```

Comment:
I think I have missed somehow in our conversation. Sorry. 
Now, I remove ```threshold_strategy``` in class ```RegionExtractor``` and use this private function for thresholding.
threshold = [float, 'auto']
default: threshold = 'auto'
```python
def _threshold_maps(maps, threshold):
    """ Automatic threshold function using ratio to the number of voxels
    in the brain volume to keep the most prominent features of the maps
    which lies above a certain threshold.
    """
    abs_maps = np.abs(maps)
    ratio = None
    if isinstance(threshold, float):
        ratio = threshold
    elif threshold == 'auto':
        ratio = 1.
    elif threshold is not None:
        raise ValueError("Threshold must be 'auto', "
                         " or float. You provided %s." %
                         str(threshold))
    raveled = abs_maps.ravel()
    argsort = np.argsort(raveled)
    n_voxels = (ratio * maps.size)
    cutoff_threshold = raveled[argsort[- n_voxels]]
    maps[abs_maps < cutoff_threshold] = 0.
    
    return maps
```
@AlexandreAbraham 
Question is to threshold with common float value on all maps or value with each individual map ?